### ISSUE TITLE
OpenMC with OpenMP on BG/Q -- does not work

### ISSUE THREAD

**Issue by tzislam** [openmc-dev/openmc#237]
Hi, 
  Please do let me know if this is not the right place to ask for help related to OpenMC. 
I have been trying to run OpenMC on BG/Q (Vulcan and Mira) with OpenMP. My focus is on studying on-node performance of OpenMC on different architectures. 
My makefile has the following:

ifeq ($(MACHINE),bluegeneq)
  F90 = mpixlf2003_r
  F90FLAGS += -WF,-DNO_F2008,-DMPI -O5
endif

NOTE: F90FLAGS has += instead of :=, which is the case in the default Makefile. This is important since when I turn on "OPENMP = yes", the F90FLAGS variable gets then overwritten. Hence, the compilation process does not really include "-qsmp=omp" string. The following job submission string on Mira work for me:

[The one that works: MPI, no OPENMP]
[Mira] qsub -n 64 --mode c1 -t 60 --env XLFRTEOPTS=ufmt_littleendian=7 ./openmc /path/to/settings.xml_file
[Vulcan] env XLFRTEOPTS=ufmt_littleendian=7 srun -n 64 ./openmc /path/to/settings.xml_file

[The one that does not work, OPENMP]
[Mira]qsub -n 1 --mode c1 -t 60 --env XLFRTEOPTS=ufmt_littleendian=7:OMP_NUM_THREADS=16 ./openmc -s 16 /path/to/settings.xml_file
[Vulcan]env XLFRTEOPTS=ufmt_littleendian=7 OMP_NUM_THREADS=16 srun -n 1 ./openmc -s 16 /path/to/settings.xml_file

In short, when I correctly compile with OpenMP, OpenMC crashes after running for sometime, specially after reading input files. The same program runs with OpenMP turned off. I followed the steps described in the User guide. 

Did anyone else successfully run OpenMC on Vulcan or Mira with OpenMP enabled?
I heard that on BG/Q nested parallelism for Fortran+OpenMP does not work, but I am not sure. Does OpenMC have nested parallelism that is possibly the cause of this crash?

Will appreciate any help.
thank you!
--Tanzima
![openmc-openmp-vulcan](https://f.cloud.github.com/assets/4874022/1874247/d4d979a4-78cb-11e3-8018-6afed21ba79d.png)
